src/main/java/seedu/taassist/model/uniquelist/UniqueList.java

```diff
@@ -44,6 +46,24 @@ public void add(T toAdd) {
             throw new DuplicateElementException();
         }
         internalList.add(toAdd);
+        Collections.sort(internalList);
+    }
+
+    /**
+     * Adds all elements in the collection to the list.
+     * The elements must not already exist in the list.
+     */
+    public void addAll(Collection<T> toAddList) {
+        requireAllNonNull(toAddList);
+        for (T toAdd : toAddList) {
+            if (contains(toAdd)) {
+                throw new DuplicateElementException();
+            }
+        }
+        for (T toAdd : toAddList) {
+            internalList.add(toAdd);
```

Comment:
Optional, but consider using `addAll` instead to reduce the number of changes that will be fired

Reply:
Great suggestion!